src/Core/Services/Implementations/OrganizationService.cs

```diff
@@ -2115,9 +2117,9 @@ private void ValidateOrganizationUpgradeParameters(Models.StaticStore.Plan plan,
             return;
         }
 
-        if (oldType == OrganizationUserType.Custom || newType == OrganizationUserType.Custom)
+        if ((oldType == OrganizationUserType.Custom || newType == OrganizationUserType.Custom) && !await _currentContext.OrganizationCustom(organizationId))
         {
-            throw new BadRequestException("Only Owners and Admins can configure Custom accounts.");
+            throw new BadRequestException("Only Owners and Admins or other Custom users can configure Custom accounts.");
         }
```

Comment:
The purpose of this check is to prevent Custom users from modifying other Custom users. We now allow that in some circumstances, so I assume this check should be deleted and replaced with a call to `ValidateCustomPermissions` instead.

That also ensures that `ValidateCustomPermissions` is called from all the code flows that call this private method. There's quite a few similar ones in this service (save, invite, updateUserGroups, etc) and it doesn't look like they're all covered by the `ValidateCustomPermissions` check at the moment.

Reply:
Good point, I modified `ValidateOrganizationUserUpdatePermissions` to call `ValidateCustomPermissionsGrant`